nemo/collections/asr/models/hybrid_rnnt_ctc_bpe_models.py

```diff
@@ -0,0 +1,185 @@
+            ctc_decoding_cfg: A config for auxiliary CTC decoding, which is optional and can be used to change the decoding type.
+
+        Returns: None
+
+        """
+        super().__init__(
+            new_tokenizer_dir=new_tokenizer_dir, new_tokenizer_type=new_tokenizer_type, decoding_cfg=decoding_cfg
+        )
+
+        # set up CTC decoder if required
+        if ctc_decoding_cfg is not None:  # or self.ctc_loss_weight > 0:
+            if hasattr(self, 'ctc_decoding'):
+                decoder_config = self.ctc_decoding.to_config_dict()
+                new_decoder_config = copy.deepcopy(decoder_config)
+
+                del self.ctc_decoding
+                del self.ctc_loss
+            else:
+                new_decoder_config = self.cfg.ctc_decoding.decoder
+
+            # sidestepping the potential overlapping tokens issue in aggregate tokenizers
+            if self.tokenizer_type == "agg":
+                new_decoder_config.vocabulary = ListConfig(vocabulary)
+            else:
+                new_decoder_config.vocabulary = ListConfig(list(vocabulary.keys()))
+            new_decoder_config['num_classes'] = len(vocabulary)
+
+            self.ctc_decoder = EncDecCTCModelBPE.from_config_dict(new_decoder_config)
+            self.ctc_loss = CTCLoss(
+                num_classes=self.ctc_decoding.num_classes_with_blank - 1,
+                zero_infinity=True,
+                reduction=self.cfg.ctc_decoding.get("ctc_reduction", "mean_batch"),
+            )
+
+            if ctc_decoding_cfg is None:
+                ctc_decoding_cfg = self.cfg.ctc_decoding.get('decoding', None)
+            if ctc_decoding_cfg is None:
+                ctc_decoding_cfg = OmegaConf.structured(CTCBPEDecodingConfig)
+                with open_dict(self.cfg.ctc_decoding):
+                    self.cfg.ctc_decoding.decoder = ctc_decoding_cfg
+
+            # Assert the decoding config with all hyper parameters
+            ctc_decoding_cls = OmegaConf.structured(CTCBPEDecodingConfig)
+            ctc_decoding_cls = OmegaConf.create(OmegaConf.to_container(ctc_decoding_cls))
+            ctc_decoding_cfg = OmegaConf.merge(ctc_decoding_cls, ctc_decoding_cfg)
+
+            self.ctc_decoding = CTCBPEDecoding(decoding_cfg=ctc_decoding_cfg, tokenizer=self.tokenizer)
+
+            self.ctc_wer = WERBPE(
+                decoding=self.ctc_decoding,
+                use_cer=self.cfg.ctc_decoding.get('use_cer', False),
+                log_prediction=self.cfg.get("log_prediction", False),
+                dist_sync_on_step=True,
+            )
+
+            # Update config
+            with open_dict(self.cfg.ctc_decoding.decoder):
+                self.cfg.ctc_decoding.decoder = new_decoder_config
+
+            with open_dict(self.cfg.ctc_decoding.decoding):
+                self.cfg.ctc_decoding.decoding = ctc_decoding_cfg
+
+    def change_decoding_strategy(self, decoding_cfg: DictConfig, decoder_type: str = None):
```

Comment:
Hmm would it be useful to have both decoding results in one shot? I mean decoder_type="both" and then give somehow return 2 Hypothesis lists - one for rnnt and other for ctc. Maybe too complicated.

 I think id prefer two separate calls to the model - easier for our inference and pipelines

Reply:
That feature can be useful, but not sure if it is easy without more duplication in the code. I would try.

Reply:
I think skip it for now, we can do it in the future.